### PR TITLE
Add integration test for dataset module

### LINKED ISSUE #267: Add new set of tests in dataset provider module to check for compilation errors with Keycloak nightly code base
### Description

Add new set of tests in dataset provider module to check for compilation errors with Keycloak nightly code base 

### Discussion

_No response_

### Motivation

Improve developer feedback by finding compile time issues with the most recent Keycloak nightly build

### Details

- Update the https://github.com/keycloak/keycloak-benchmark/blob/main/pom.xml#L13 Keycloak pom with the recent nightly build pulled from a repository
- Use a light weight provisioning module to bring up Keycloak and a Database
- Write new smoke tests to make sure things compile and we can deploy the dataset module to the Keycloak and perform some dataset creation operations.

### PR DESCRIPTION
**Pull request by ahus1**
Also adding a scheduled build every week.

Closes #267

### CODE REVIEW

**Comment by ahus1** [comment]
Hi @kami619 - here's now an integration tests which only uses Java please let me know what you think of this.

**Comment by ahus1** [comment]
side note: with these tests in place, I wonder if we still need all the tests in `provision-minikube.yml`, or if we could skip some of them as we don't expect them to reveal many additional errors.

**Comment by kami619** [comment]
@ahus1 I will get to this today, sorry been occupied with other tasks

File: doc/dataset/modules/ROOT/pages/changing-provider.adoc [hunk 1]
```diff
@@ -18,7 +18,18 @@ The recommended setup is to first complete the guide xref:installation/installat
 
 Developing the provider against Keycloak Wildfly, or with a locally running instance of Keycloak Quarkus is possible as well, still it is not part of this guide today.
 
-== Preparing changes to the provider
+== Preparing changes using the Minikube setup
```
Comment (kami619) [inline doc/dataset/modules/ROOT/pages/changing-provider.adoc:21]: Should this be `Preparing changes using the local setup` of sorts ? @ahus1

File: dataset/src/test/java/org/keycloak/benchmark/it/DeploymentIT.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    }
+
+    @Test
+    public void deployWithLegacyStore() throws IOException, ExecutionException, InterruptedException, URISyntaxException {
+        runTestWithParameter();
+    }
+
+    @Test
+    public void deployWithMapStore() throws IOException, ExecutionException, InterruptedException, URISyntaxException {
+        runTestWithParameter("--storage", "chm");
```
Comment (kami619) [inline dataset/src/test/java/org/keycloak/benchmark/it/DeploymentIT.java:93]: @ahus1 does this take values such as ```runTestWithParameter("--storage", "hotrod")``` ? If I create a new test with this option and run, it currently fails

File: doc/dataset/modules/ROOT/pages/changing-provider.adoc (hunk 1 above)
Comment (ahus1) [reply doc/dataset/modules/ROOT/pages/changing-provider.adoc]: Oops, a copy-and-paste error! I pushed a change to correct this in the way you suggested.

**Comment by kami619** [comment]
> side note: with these tests in place, I wonder if we still need all the tests in `provision-minikube.yml`, or if we could skip some of them as we don't expect them to reveal many additional errors.

I would still keep those tests from `provision-minikube.yml` as they cover external databases such as PostgreSQL and CockroachDB, just in case we have any integration issues with them.

**Pull request merged**